digest/digest_test.go

```diff
@@ -14,21 +12,6 @@ func TestParseDigest(t *testing.T) {
 		hex       string
 	}{
 		{
-			input:     "tarsum+sha256:e58fcf7418d4390dec8e8fb69d88c06ec07039d651fedd3aa72af9972e7d046b",
-			algorithm: "tarsum+sha256",
```

Comment:
Why can't `hash.Hash` support this? Are saying because of the symbol? I think we'd still have to unroll each algorithm.